### DISCUSSION TITLE
Bug in query handling with multiple values

### DISCUSSION THREAD

**Discussion by ed2050** [fastapi/fastapi#10724] [Questions] [answered]
### First Check

- [X] I added a very descriptive title here.
- [X] I used the GitHub search to find a similar question and didn't find it.
- [X] I searched the FastAPI documentation, with the integrated search.
- [X] I already searched in Google "How to X in FastAPI" and didn't find any information.
- [X] I already read and followed all the tutorial in the docs and didn't find an answer.
- [X] I already checked if it is not related to FastAPI but to [Pydantic](https://github.com/pydantic/pydantic).
- [X] I already checked if it is not related to FastAPI but to [Swagger UI](https://github.com/swagger-api/swagger-ui).
- [X] I already checked if it is not related to FastAPI but to [ReDoc](https://github.com/Redocly/redoc).

### Commit to Help

- [X] I commit to help with one of those options 👆

### Example Code

```python
from fastapi import FastAPI

app = FastAPI()

@app.get("/test")
async def mypage (name = None) :  # get `name` from query params
    print ('{ name = }')


1. go to server/test?name=foo : prints "name = foo" OK
2. go to server/test?name=foo&name=bar : prints "name = bar" INCORRECT

Per python stdlib and URL DOM standard, second item should print `name = ["foo", "bar"]`.
```

### Description

There's a problem with nicegui's query param implementation.  Last instance of query param shadows previous values.  In contrast, python stdlib returns an array of values for each param.

Python stdlib behavior (also behavior of [URL DOM standard](https://url.spec.whatwg.org/#dom-urlsearchparams-getall)):
```py
from urllib.parse import parse_qs

parse_qs ('name=foo')
# returns {'name': ['foo']}

parse_qs ('name=foo&name=bar')
# returns {'name': ['foo', 'bar']}
```

Standard behavior creates an array for each query param so multiple values are retained.  FastAPI returns last value only, which is nonstandard behavior.

Note that DOM returns the _first_  value if only one is requested.  Here, since users can't choose between one or multiple, it would be better for FastAPI to return all values.  Either way, returning only last value is out of scope.

### Operating System

macOS

### FastAPI Version

0.103.1

### Pydantic Version

2.3.0

### Python Version

3.8.2

**Comment by JavierSanchezCastro** [comment, marked as answer]
https://fastapi.tiangolo.com/tutorial/query-params-str-validations/#query-parameter-list-multiple-values

Is this what you are looking for?
